doc/tutorials/tutorial_understanding_building_benchmark.rst

```diff
@@ -0,0 +1,370 @@
+                beta -= alpha * gradient_ols(self.X, self.y, beta)
+
+            self.beta = beta
+
+
+    - the ``get_data`` method in ``simulated.py`` where \\(y \\) and \\(X \\) are defined. In this template benchmark, they are simply generated randomly using numpy.
+
+    .. code-block:: python
+
+        def get_data(self):
+            rng = np.random.RandomState(self.random_state)
+            X = rng.randn(self.n_samples, self.n_features)
+            y = rng.randn(self.n_samples)
+            return dict(X=X, y=y),
+
+The second type of methods found in these three python files are the **communication** tools.
+Indeed, solvers, dataset and objectives need to exchange information.
+Typically, the solver needs to know the parameters used for the loss, in our case the value of the regularization parameter.
+The objective needs to know the values of \\( X\\) and \\( y\\) defined in the dataset.
+This part of the benchmark can rapidly give you a headache if you are not familiar with the structure of the benchmark, but the figure above should be a good reference point.
+When a method from a class feeds a method in another class, it returns a dictionary (such as ``get_data`` we just discussed), otherwise it simply updates the class attributes.
+
+.. The previous figure may be updated to detail these communication methods.
+
+Finally, one may wonder where to define the hyperparameters of the problem.
+The general rule of thumb is that hyperparameters are defined as attributes of solvers, objectives or dataset depending on where it makes the most sense.
+For instance the stepsize is a solver-dependent parameter, it is defined as an attribute of the ``python-gd`` solver
+
+.. prompt:: python
+
+    class Solver(BaseSolver):
+        name="GD"
+        parameters = {
+            'scale_step': [1, 1.99]
+        }
+
+Updating the template to implement a ridge regression benchmark
+---------------------------------------------------------------
+
+We are now equipped with enough knowledge to update the template benchmark to a ridge regression benchmark.
+Formally, we are starting from OLS and GD implemented for the OLS problem.
+Therefore we need to implement the following modifications:
+
+    - we should add the regularization term \\( +\\lambda \\|\\beta \\|^2 \\) to the loss in ``objective.py``, and values for the regularization parameter.
+    - we should modify the computed gradient, knowing that \\( \\nabla f(\\beta) = \\nabla g(\\beta) + 2\\lambda\\beta \\).
+
+We will not modify anything in the dataset since the inputs \\(X,y \\) of the regression and ridge regression are essentially the same.
+
+Let's start with the ``objective.py`` file.
+The regularization parameter values are part of the formal definition of the problem, so we can define them as attributes of the ``Objective`` class.
+The ``whiten_y`` parameter is already implemented, so we can simply add a ``reg`` parameter in the ``parameters`` dictionary.
```

Comment:
we can also **replace** the whiten parameter, no?

Reply:
Ok I removed the True option for simplicity